lib/srv/db/proxyserver.go

```diff
@@ -529,7 +530,23 @@ func (s *ProxyServer) Proxy(ctx context.Context, proxyCtx *common.ProxyContext,
 	activeConnections.With(labels).Inc()
 	defer activeConnections.With(labels).Dec()
 
-	return trace.Wrap(utils.ProxyConn(ctx, clientConn, serviceConn))
+	err = utils.ProxyConn(ctx, clientConn, serviceConn)
+
+	// Ignore context.Canceled errors from monitoring.
+	//
+	// The clientConn is closed by utils.ProxyConn on successful io.Copy thus
+	// possibly causing utils.ProxyConn to return context.Canceled (as monitor
+	// context is closed when TrackingReadConn.Close() is called).
+	//
+	// In cases MonitorConn kills the connection for other reasons (e.g.
+	// expired cert), a proper cause will be returned instead of
+	// context.Canceled.
+	if trace.Unwrap(err) == context.Canceled {
```

Comment:
can we use `io.EOF` instead of `context.Canceled`?
Feels like a much better usecase.
Can you also use `errors.Is` instead of `trace.Unwrap`?